dubbo-plugin/dubbo-mcp/src/main/java/org/apache/dubbo/mcp/core/McpServiceFilter.java

```
@@ -0,0 +1,383 @@
+        if (serviceUrl != null) {
+            String methodUrlValue = serviceUrl.getMethodParameter(methodName, McpConstant.PARAM_MCP_ENABLED);
+
+            if (methodUrlValue != null && StringUtils.isNotEmpty(methodUrlValue)) {
+                return Boolean.parseBoolean(methodUrlValue);
+            }
+
+            String serviceLevelValue = serviceUrl.getParameter(McpConstant.PARAM_MCP_ENABLED);
+
+            if (serviceLevelValue != null && StringUtils.isNotEmpty(serviceLevelValue)) {
+                return Boolean.parseBoolean(serviceLevelValue);
+            }
+        }
+
+        if (methodMcpTool != null && methodMcpTool.enabled()) {
+            return true;
+        }
+
+        String methodConfigKey =
+                McpConstant.SETTINGS_MCP_SERVICE_PREFIX + "." + interfaceName + ".methods." + methodName + ".enabled";
+        if (configuration.containsKey(methodConfigKey)) {
+            return configuration.getBoolean(methodConfigKey, false);
+        }
+
+        if (shouldExposeAsMcpTool(providerModel)) {
+            return Modifier.isPublic(method.getModifiers());
+        }
+
+        return false;
+    }
+
+    /**
+     * Get @McpTool annotation from method, checking both interface and implementation class.
+     */
+    private McpTool getMethodMcpTool(ProviderModel providerModel, Method method) {
+        String methodName = method.getName();
+        Class<?>[] paramTypes = method.getParameterTypes();
+
+        Object serviceBean = providerModel.getServiceInstance();
+        if (serviceBean != null) {
+            try {
+                Method implMethod = serviceBean.getClass().getMethod(methodName, paramTypes);
+                McpTool implMcpTool = implMethod.getAnnotation(McpTool.class);
+                if (implMcpTool != null) {
+                    return implMcpTool;
+                }
+            } catch (NoSuchMethodException e) {
+                // Method not found in implementation class
```

don't ignore exception

---

Done